src/__tests__/displayAll.test.js

```diff
@@ -0,0 +1,32 @@
+/* eslint-disable  */
+import React from 'react';
+
+import { render, screen } from '@testing-library/react';
+import * as router from 'react-router';
+import { Provider } from 'react-redux';
+import store from '../redux/Store';
+import DisplayAll from '../components/displayAll';
+import StockExchangeData from '../components/stockExchangeData';
+
+
+
+
+describe('expected to get a correct output', () => {
+    const Navigation = jest.fn();  
+    beforeEach(() => {
+      jest.spyOn(router, 'useNavigate').mockImplementation(() => Navigation);
+    });
+    
+    test('it must render the page for particulars', () => {
+      render(<Provider store={store}><StockExchangeData /></Provider>);
+      expect(screen.findAllByText('Price change(usd):')).toMatchSnapshot();
+    });
+    test('must render the home page', () => {
+        render(<Provider store={store}><DisplayAll /></Provider>);
+        expect(screen.findAllByText(' Crypto currency market data')).toMatchSnapshot();
+      });
+   
+
+   
+  });
+  
```

Comment:
- [ ] as required you should add more tests for other components like stock exchange and wrapper in order to complete the requirements 
- [ ] and also you need to write tests for redux pure functions as required in order to complete the requirements 
<img width="673" alt="Screenshot 2022-11-30 at 17 23 07" src="https://user-images.githubusercontent.com/61790587/205038554-42d9807e-5e6d-41b8-b504-f365fa4fb411.png">

Reply:
add unit tests